packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -734,3 +714,76 @@ async function areFoldersEmpty(folderPaths: string[]): Promise<boolean> {
 async function waitSeconds(second: number) {
   return new Promise((resolve) => setTimeout(resolve, second * 1000));
 }
+
+async function getDeploymentError(
+  deployCtx: DeployContext,
+  resourceGroupName: string,
+  deploymentName: string
+): Promise<any> {
+  let deployment;
+  try {
+    deployment = await deployCtx.client.deployments.get(resourceGroupName, deploymentName);
+  } catch (error) {
+    return undefined;
+  }
+  if (!deployment.properties?.error) {
+    return undefined;
+  }
+  const deploymentError: any = {
+    error: deployment.properties?.error,
+  };
+  const operations = await deployCtx.client.deploymentOperations.list(
+    resourceGroupName,
+    deploymentName
+  );
+  for (const operation of operations) {
+    if (operation.properties?.statusMessage?.error) {
+      if (!deploymentError.subErrors) {
+        deploymentError.subErrors = {};
+      }
+      const name = operation.properties.targetResource?.resourceName ?? operation.id;
+      deploymentError.subErrors[name!] = {
+        error: operation.properties.statusMessage.error,
+      };
+      if (
+        operation.properties.targetResource?.resourceType ===
+          ConstantString.DeploymentResourceType &&
+        operation.properties.targetResource?.resourceName &&
+        operation.properties.targetResource?.id
+      ) {
+        const resourceGroupName: string = getResourceGroupFromDeploymentId(
+          operation.properties.targetResource.id
+        );
+        const subError = await getDeploymentError(
```

Comment:
the resource group name from `getResourceGroupFromDeploymentId` may be empty string, maybe we need to handle this case? by the way, when will `operation.properties.targetResource.id` be null?

Reply:
It is fixed. Use the common function to replace it. If operation.properties.targetResource.id is null, throw an error.